apps/web/pages/index.tsx

```diff
@@ -147,18 +124,18 @@ const Home: NextPage = () => {
         }
       }
     }
-  }, [
```

Comment:
We're utilizing memoization hooks like `useCallback` and `useMemo` extensively, which is great for performance optimization. However, I want to highlight a potential technical debt issue arising from this practice. Tracking dependencies for these hooks can get complex, especially when there are many dependencies. 

Also, using non-primitive values in dependencies can undermine the effectiveness of memoization since React does shallow comparisons, not deep, in dependency arrays for hooks like useEffect. This could lead to unnecessary re-renders or missed updates, affecting both performance and maintainability. 

It’s crucial we strike a balance - leveraging memoization for performance while ensuring our dependency arrays are simple and efficient to maintain the overall code health.

So, I propose a cautious approach: let's reserve memoization for scenarios where it's absolutely necessary. Before turning to memoization, we should first explore optimizing our state management. This could involve refining how we handle states and updates, particularly in response to w3i API interactions.

Reply:
Memoization here wasn't strictly about performance, it was about preventing unncessary re-renders. Let's see how this plays out in testing